### DISCUSSION TITLE
Ability to display deprecation notices or other general information on a tool

### DISCUSSION THREAD

**Discussion by natefoo** [galaxyproject/galaxy#17951] [Ideas]
From a discussion on wg-ui-ux:

> @natefoo Over on .org we'd like to deprecate some tools (e.g. Trinity) while not hiding them, but discouraging their use. We can of course add tags, but it'd be great if we could display some text at the top of them pointing them at alternatives and maybe some trainings that use other tools.

@wm75 suggested using help forum integration (https://github.com/galaxyproject/galaxy/pull/16798), which seems like a great idea, although these appear at the bottom of the form, rather than the top. We could curate what would be found and displayed by using tags:

> @wm75 Currently you can't, but the whole discovery process is controlled via tags anyway, so shouldn't be hard to add a few more for setting priority.  
> and maybe the tool interface could, at the top, alert users if a tool has high-priority help available?

As Wolfgang points out, one nice benefit of using the help forum integration is that this would benefit more than just a single instance.

I'm opening a discussion for two reasons:

1. Does this seem like a good or bad idea? Any other ideas?  and,
2. How should this work on the tool form, if we want to display something more prominently?

**Comment by bgruening** [comment]
A few things that we are currently discussing are:

* failing jobs with an appropriate error message, e.g. you can use `blastp` on tiny files, but we will bail out on large files and recommend `diamond blastp` to our users
* using the new notification system after the fact ... e.g. sending notifications like "we have seen you have been using tool X on Y, but we recommend you to use Z instead because its ..." 
* use more features of the tool recommendation system in Galaxy ... tweaking recommendations to nudge people into more recent tools

Two tools where we can try Wolfgangs approach easily and maybe others is prokka/bakta and tophat, for both we know that the upstream developers recommend other tools nowadays.

**Comment by sanjaysrikakulam** [comment]
Last year, I came across this super cool [webhook](https://github.com/usegalaxy-au/infrastructure/pull/2212) implemented by @neoformit, which allows us to display a custom message for tools. I want to deploy and test this soon in the EU.

**Reply by neoformit** [reply]
FYI I had to make some updates to our webhooks for galaxy 24.2 compatibility. Seems that the new masthead has changed the rendering of these elements. See https://github.com/galaxyproject/galaxy/issues/19572

Our versioned webhooks: https://github.com/usegalaxy-au/infrastructure/tree/master/templates/galaxy/webhooks

**Reply by sanjaysrikakulam** [reply]
We have [deployed it](https://github.com/usegalaxy-eu/infrastructure-playbook/pull/1413), and it works great! Thank you @neoformit! :)

![image](https://github.com/user-attachments/assets/c3a322ff-e36a-4f7b-b91d-f4839e00f6b5)

**Reply by bernt-matthias** [reply]
Cool. Is there be a global yaml config that can be shared across instances?

**Reply by sanjaysrikakulam** [reply]
Not at the moment. If the community shows enough interest, we could build one similar to the TPV shared DB and retire/deprecate some tools collectively. 

(Just an idea) Alternately, we could,
1. Add a `message` and `class` in TPV `tools.yml` and/or TPV shared DB for each tool (to whichever ones we collectively want to retire/deprecate)
2. Parse this TPV file when templating during the deployment to construct the format required by the webhook to have one definition where everything tool-related is defined/configured.

**Reply by natefoo** [reply]
This is really nice, I've [done the same](https://github.com/galaxyproject/usegalaxy-playbook/commit/15702b1995aa143b098c05b7d1e22b27ba09072d). Thanks @neoformit! And @sanjaysrikakulam for bringing it to my attention.

**Comment by bernt-matthias** [comment]
Slightly related https://github.com/galaxyproject/galaxy/discussions/19053
